pkg/operator/defaultscccontroller/defaultscccontroller.go

```diff
@@ -0,0 +1,141 @@
+package defaultscccontroller
+
+import (
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	securityv1 "github.com/openshift/api/security/v1"
+	securityv1client "github.com/openshift/client-go/security/clientset/versioned/typed/security/v1"
+	securityv1listers "github.com/openshift/client-go/security/listers/security/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	ControllerName = "default-scc-upgradeable"
+)
+
+type Options struct {
+	Config         *rest.Config
+	OperatorClient v1helpers.OperatorClient
+	Recorder       events.Recorder
+	ResyncPeriod   time.Duration
+}
+
+func NewDefaultSCCController(options *Options) (controller *DefaultSCCController, err error) {
+	client, clientErr := securityv1client.NewForConfig(options.Config)
+	if clientErr != nil {
+		err = fmt.Errorf("[%s] failed to create client for security/v1 - %s", ControllerName, clientErr.Error())
```

Comment:
this style of named error is pretty uncommon, and IMO hard to read.

Reply:
I don't have any strong opinion either way, but what drove me to this style is:
- avoid re-purposing or reassigning `err`, wanted to avoid this 
```
err = fmt.Errorf("message - %s", err.Error())
```
The error returned from a function invocation and the error we return to the caller are separate.

- with separate names it is explicit to the reader how the error is being chained, also it is explicit that there is no shadowing. 

But again, readability is in the eye of the reader :), so if this is confusing or hard to read I can use `err` across, let me know.

Reply:
Reassignment of `err` is the usual pattern. When code does not follow it, it's suspicious as there must be a reason, making the reading flow worse.